spanner/api/Spanner/Program.cs

```diff
@@ -1343,6 +1378,177 @@ await connection.CreateDdlCommand($"DROP TABLE {table}")
             return ExitCode.Success;
         }
 
+        public static object BatchInsertRecords(string projectId,
+            string instanceId, string databaseId)
+        {
+            var response =
+                BatchInsertRecordsAsync(projectId, instanceId, databaseId);
+            Console.WriteLine("Waiting for operation to complete...");
+            response.Wait();
+            Console.WriteLine($"Operation status: {response.Status}");
+            Console.WriteLine($"Inserted records into sample database "
+                + $"{databaseId} on instance {instanceId}");
+            return ExitCode.Success;
+        }
+
+        private static async Task BatchInsertRecordsAsync(string projectId,
+            string instanceId, string databaseId)
+        {
+            string connectionString =
+                $"Data Source=projects/{projectId}/instances/{instanceId}"
+                + $"/databases/{databaseId}";
+
+            // Get current max SingerId.
+            Int64 maxSingerId = 0;
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                // Execute a SQL statement to get current MAX() of SingerId.
+                var cmd = connection.CreateSelectCommand(
+                    @"SELECT MAX(SingerId) as SingerId FROM Singers");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        long parsedValue;
+                        bool result = Int64.TryParse(
+                            reader.GetFieldValue<string>("SingerId"), out parsedValue);
+                        if (result) maxSingerId = parsedValue;
+                    }
+                }
+            }
+
+            // Batch insert 249,900 singer records into the Singers table.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                await connection.OpenAsync();
+
+                for (int i = 0; i < 100; i++)
+                {
+                    using (var tx = await connection.BeginTransactionAsync())
+                    using (var cmd = connection.CreateInsertCommand("Singers", new SpannerParameterCollection
+                    {
+                        {"SingerId", SpannerDbType.String},
```

Comment:
nit: please, add one space after { and before }

Reply:
Done.